lib/class-wp-rest-menu-items-controller.php

```diff
@@ -671,7 +671,7 @@ public function prepare_item_for_response( $post, $request ) {
 
 		if ( rest_is_field_included( 'parent', $fields ) ) {
 			// Same as post_parent, expose as integer.
-			$data['parent'] = (int) $menu_item->post_parent;
+			$data['parent'] = (int) $menu_item->menu_item_parent;
```

Comment:
Without this change, the API always returns parent ID = 0 even though it sets it correctly on PUT. Let's revisit the naming of these fields in another PR.

Reply:
This is fixed in trunk. I recommend rebasing. We fixed a couple of things on the rest api.

Reply:
See https://github.com/WordPress/gutenberg/pull/34835